environment/src/main/java/com/sequenceiq/environment/environment/validation/securitygroup/azure/AzureEnvironmentSecurityGroupValidator.java

```diff
@@ -59,18 +60,24 @@ private void validateSecurityGroup(EnvironmentDto environmentDto, ValidationResu
         CloudSecurityGroups securityGroups = platformParameterService.getSecurityGroups(request);
 
         boolean securityGroupFoundInRegion = false;
-        if (Objects.nonNull(securityGroups.getCloudSecurityGroupsResponses())
-                && Objects.nonNull(securityGroups.getCloudSecurityGroupsResponses().get(region.getName()))) {
-            for (CloudSecurityGroup cloudSecurityGroup : securityGroups.getCloudSecurityGroupsResponses().get(region.getName())) {
-                String groupId = cloudSecurityGroup.getGroupId();
-                if (groupId.equalsIgnoreCase(securityGroupId)) {
-                    securityGroupFoundInRegion = true;
-                    break;
+        for (String securityGroupId : getSecurityGroupIds(securityGroupIds)) {
+            securityGroupFoundInRegion = false;
+            if (Objects.nonNull(securityGroups.getCloudSecurityGroupsResponses())
+                    && Objects.nonNull(securityGroups.getCloudSecurityGroupsResponses().get(region.getName()))) {
+                for (CloudSecurityGroup cloudSecurityGroup : securityGroups.getCloudSecurityGroupsResponses().get(region.getName())) {
```

Comment:
Nit: `securityGroups.getCloudSecurityGroupsResponses().get(region.getName())` does not change, so it would make sense to extract it into a local variable outside the `getSecurityGroupIds()` loop of line 80 (but still inside the `if(nonNull()...)` block).